app/scripts/lib/util.js

```diff
@@ -139,6 +139,17 @@ const addHexPrefix = (str) => {
   return `0x${str}`;
 };
 
+const fixHexValue = (str) => {
+  const prefixed = addHexPrefix(str);
+  // if it's a valid hex return it immediately
+  if (prefixed.match(/^-?0x[a-fA-F0-9]+$/u)) {
+    return prefixed;
+  }
+
+  // Else return 0x0 for all non valid hex values
+  return '0x0';
```

Comment:
I'm okay with this as it, but just throwing it out there that maybe we shouldn't evaluate all improper formatted values to '0x0'. We may want to throw an error and literally only fix 0x to 0x0. @Gudahtt thoughts?

Reply:
Why are we even fixing this case? According to the Ethereum JSON-RPC docs, "0x" is not a valid value for the transaction value. See the definition of "QUANTITY".
https://eth.wiki/json-rpc/API

We should return an error to the dapp if they give us invalid parameters.

Reply:
We can throw the error in the same file, lower in the validation pieces. It won't make it to the point where the big number error happens if we do this resolving the original bug and improving error messaging for dapps. I think that's a win-win